tests/installation/test_pip_installer.py

```diff
@@ -204,3 +206,32 @@ def copy_only(source: Path, dest: Path) -> None:
     # any command in the virtual environment should trigger the error message
     output = tmp_venv.run("python", "-m", "site")
     assert not re.match(rf"Error processing line 1 of .*{pth_file}", output)
+
+
+def test_install_with_trusted_host(config: Config):
+    config.merge({"certificates": {"default": {"cert": False}}})
+
+    default = LegacyRepository("default", "https://foo.bar")
+    pool = Pool()
+    pool.add_repository(default, default=True)
+
+    null_env = NullEnv()
+
+    installer = PipInstaller(null_env, NullIO(), pool)
+
+    foo = Package(
+        "foo",
+        "0.0.0",
+        source_type="legacy",
+        source_reference=default.name,
+        source_url=default.url,
+    )
+
+    installer.install(foo)
+
+    assert len(null_env.executed) == 1
+    cmd = null_env.executed[0]
+    assert "--trusted-host" in cmd
+    cert_index = cmd.index("--trusted-host")
+    # Need to do the str(Path()) bit because Windows paths get modified by Path
```

Comment:
Does this comment belong here? I don't see such code anywhere around

Reply:
This is an orphan after the fix for the failing Windows tests -- good eye.

Reply:
It's copied over from #3676, will remve it.